File: app/src/main/java/com/hjc/base/model/CommonModel.java
package com.hjc.base.model;

import com.hjc.library_base.model.BaseModel;
import com.hjc.library_net.SmartHttp;

/**
 * @Author: HJC
 * @Date: 2021/9/22 20:24
 * @Description: 通用Model
 */
public class CommonModel extends BaseModel {

    public <T> T getApiService(Class<T> apiService) {
        return SmartHttp.getInstance().getRetrofit().create(apiService);
    }

}

File: library-common/src/main/java/com/hjc/library_common/model/CommonModel.kt
package com.hjc.library_common.model

import com.hjc.library_base.model.BaseModel
import com.hjc.library_net.SmartHttp


/**
 * @Author: HJC
 * @Date: 2020/6/5 16:44
 * @Description: 通用Model
 */
open class CommonModel : BaseModel() {

    fun <T> getApiService(apiService: Class<T>): T {
        return SmartHttp.getRetrofit().create(apiService)
    }
}